cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/AbstractDistributedProgramRunner.java

```diff
@@ -157,19 +168,46 @@ protected EventHandler createEventHandler(CConfiguration cConf) {
     return new AbortOnTimeoutEventHandler(cConf.getLong(Constants.CFG_TWILL_NO_CONTAINER_TIMEOUT, Long.MAX_VALUE));
   }
 
+  // returns the path of the file, if its a local URI. If it's an HDFS URI, copy it to local file system first.
+  private String getKeytabPath(URI keytabURI, File tempDir) throws IOException {
+    if (!"hdfs".equals(keytabURI.getScheme())) {
+      return keytabURI.getPath();
```

Comment:
I think we can support it, if LocationFactory can support it. Specifically, if FileContextLocationFactory can support it.